crates/metadata-providers/src/objstore/optimistic_store.rs

```diff
@@ -89,15 +118,33 @@ impl OptimisticLockingMetadataStore {
 
     fn serialize_versioned_value(
         &mut self,
-        versioned_value: &VersionedValue,
-        cookie: u64,
-    ) -> Result<Bytes, WriteError> {
-        self.arena.clear();
-        let writer = (&mut self.arena).writer();
-        let on_disk = OnDiskValue::V1(Cow::Borrowed(versioned_value), cookie);
-        ciborium::into_writer(&on_disk, writer)
-            .map(|_| self.arena.split().freeze())
-            .map_err(|e| WriteError::Codec(e.into()))
+        encoding: ValueEncoding,
+        versioned_value: VersionedValue,
+        salt: u64,
+    ) -> Result<Content, WriteError> {
+        match encoding {
+            ValueEncoding::Cbor => {
+                self.arena.clear();
+                let writer = (&mut self.arena).writer();
+                let on_disk = OnDiskValue::V1(Cow::Owned(versioned_value), salt);
+                ciborium::into_writer(&on_disk, writer)
+                    .map(|_| self.arena.split().freeze())
+                    .map_err(|e| WriteError::Codec(e.into()))
+            }
+            ValueEncoding::Bilrost => {
+                self.arena.clear();
+                let salted = SaltedVersionedValue {
+                    salt,
+                    value: versioned_value,
+                };
+
+                salted
+                    .encode(&mut self.arena)
```

Comment:
Should we reserve() the expected length first from the arena? In other words, how sure are we that the arena is big enough to encode the value?

Reply:
That's a really good catch. Thank you for pointing this out. the arena had enough capacity for the tests, but would fail eventually.